src/ControlSurface/Controls/Colorpicker/Colorpicker.qml

```diff
@@ -0,0 +1,185 @@
+//  A toy QML colorpicker control, by Ruslan Shestopalyuk
+import QtQuick 2.11
+import QtQuick.Layouts 1.1
+import QtQuick.Controls 2.4
+
+//import "content"
+
+Rectangle {
+    id: picker
+    property color colorValue: paletteMode ? _rgb(paletts.paletts_color,
+                                          alphaSlider.value) : _hsla(
+                                         hueSlider.value,
+                                         sbPicker.saturation,
+                                         sbPicker.brightness,
+                                         1)
+    property bool enableAlphaChannel: true
+    property bool enableDetails: true
+    property int colorHandleRadius: 8
+    property bool paletteMode: false
+    property bool enablePaletteMode: false
+    property string switchToColorPickerString: "Palette..."
+    property string switchToPalleteString: "Color Picker..."
+
+    signal colorChanged(color changedColor)
+
+    clip: true
+
+    color: "#363636"
+
+    SBPicker {
+        id: sbPicker
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        anchors.left: parent.left
+        width: colorpicker.width * (83 / 100)
+
+        hueColor: {
+            var v = 1.0 - hueSlider.value
+
+            //console.debug("v:"+v)
+            if (0.0 <= v && v < 0.16) {
+                return Qt.rgba(1.0, 0.0, v / 0.16, 1.0)
+            } else if (0.16 <= v && v < 0.33) {
+                return Qt.rgba(1.0 - (v - 0.16) / 0.17, 0.0, 1.0, 1.0)
+            } else if (0.33 <= v && v < 0.5) {
+                return Qt.rgba(0.0, ((v - 0.33) / 0.17), 1.0, 1.0)
+            } else if (0.5 <= v && v < 0.76) {
+                return Qt.rgba(0.0, 1.0, 1.0 - (v - 0.5) / 0.26, 1.0)
+            } else if (0.76 <= v && v < 0.85) {
+                return Qt.rgba((v - 0.76) / 0.09, 1.0, 0.0, 1.0)
+            } else if (0.85 <= v && v <= 1.0) {
+                return Qt.rgba(1.0, 1.0 - (v - 0.85) / 0.15, 0.0, 1.0)
+            } else {
+                //console.log("hue value is outside of expected boundaries of [0, 1]")
+                return "red"
+            }
+        }
+    }
+
+    // hue picking slider
+    Item {
+        id: huePicker
+        visible: !paletteMode
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        anchors.right: parent.right
+        width: colorpicker.width * (15 / 100)
+
+        anchors.leftMargin: 5
+        anchors.rightMargin: 5
+        Layout.fillHeight: true
+        anchors.topMargin: colorHandleRadius
+        anchors.bottomMargin: colorHandleRadius
+
+        Rectangle {
+            id: colorBar
+            anchors.fill: parent
+
+            gradient: Gradient {
+                GradientStop {
+                    position: 1.0
+                    color: "#FF0000"
```

Comment:
Not super important but in Qt, the hex colors are a bit slower than e.g. https://doc.qt.io/qt-5/qml-qtqml-qt.html#rgba-method